cfgov/jobmanager/tests/test_blocks.py

```diff
@@ -74,11 +75,16 @@ def test_database_queries(self):
         for i in range(5):
             self.make_job(f'live{i}')
 
-        # We expect three database queries here. First, Wagtail has to look up
+        # We expect four database queries here. First, Wagtail has to look up
         # the site root paths. These get cached on the request object. Then,
         # all of the JobListingPages are retrieved in a single query. Finally,
         # another query retrieves the URL for the "more jobs page" link.
-        with self.assertNumQueries(3):
+        if wagtail.VERSION > (2, 6):
+            num_queries = 4
+        else:
+            num_queries = 3
```

Comment:
@willbarton I appreciate your review comment. I agree that the comment should be updated.

@chosak Do you know why the number of database queries increased when I upgraded to Wagtail 2.9?

I was curious to know why the jobmanager block tests check for the number of queries to the database.
These are the 4 Captured queries:
```
1. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."live_revision_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."id" = 3
2. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."live_revision_id", "v1_cfgovpage"."page_ptr_id", "v1_cfgovpage"."language", "v1_cfgovpage"."social_sharing_image_id", "v1_cfgovpage"."schema_json", "v1_cfgovpage"."force_breadcrumbs", "v1_cfgovpage"."sidefoot", "jobmanager_joblistingpage"."cfgovpage_ptr_id", "jobmanager_joblistingpage"."description", "jobmanager_joblistingpage"."open_date", "jobmanager_joblistingpage"."close_date", "jobmanager_joblistingpage"."salary_min", "jobmanager_joblistingpage"."salary_max", "jobmanager_joblistingpage"."division_id", "jobmanager_joblistingpage"."job_length_id", "jobmanager_joblistingpage"."service_type_id", "jobmanager_joblistingpage"."allow_remote", "jobmanager_joblistingpage"."responsibilities", "jobmanager_joblistingpage"."travel_required", "jobmanager_joblistingpage"."travel_details", "jobmanager_joblistingpage"."additional_section_title", "jobmanager_joblistingpage"."additional_section_content" FROM "jobmanager_joblistingpage" INNER JOIN "v1_cfgovpage" ON ("jobmanager_joblistingpage"."cfgovpage_ptr_id" = "v1_cfgovpage"."page_ptr_id") INNER JOIN "wagtailcore_page" ON ("v1_cfgovpage"."page_ptr_id" = "wagtailcore_page"."id") WHERE ("wagtailcore_page"."live" = true AND "jobmanager_joblistingpage"."open_date" <= '2020-07-01'::date AND "jobmanager_joblistingpage"."close_date" >= '2020-07-01'::date) ORDER BY "jobmanager_joblistingpage"."close_date" ASC, "wagtailcore_page"."title" ASC  LIMIT 5
3. SELECT "wagtailcore_site"."id", "wagtailcore_site"."hostname", "wagtailcore_site"."port", "wagtailcore_site"."site_name", "wagtailcore_site"."root_page_id", "wagtailcore_site"."is_default_site", CASE WHEN ("wagtailcore_site"."hostname" = 'testserver' AND "wagtailcore_site"."port" = 80) THEN 0 WHEN ("wagtailcore_site"."hostname" = 'testserver' AND "wagtailcore_site"."is_default_site" = true) THEN 1 WHEN "wagtailcore_site"."is_default_site" = true THEN 2 ELSE 3 END AS "match", "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."live_revision_id" FROM "wagtailcore_site" INNER JOIN "wagtailcore_page" ON ("wagtailcore_site"."root_page_id" = "wagtailcore_page"."id") WHERE ("wagtailcore_site"."hostname" = 'testserver' OR "wagtailcore_site"."is_default_site" = true) ORDER BY "match" ASC
4. SELECT "wagtailcore_site"."id", "wagtailcore_site"."hostname", "wagtailcore_site"."port", "wagtailcore_site"."site_name", "wagtailcore_site"."root_page_id", "wagtailcore_site"."is_default_site", "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."live_revision_id" FROM "wagtailcore_site" INNER JOIN "wagtailcore_page" ON ("wagtailcore_site"."root_page_id" = "wagtailcore_page"."id") ORDER BY "wagtailcore_page"."url_path" DESC, "wagtailcore_site"."is_default_site" DESC, "wagtailcore_site"."hostname" ASC
```

Reply:
The comment below needs to be updated to include what the new fourth query is:
```
        # We expect four database queries here. First, Wagtail has to look up
        # the site root paths. These get cached on the request object. Then,
        # all of the JobListingPages are retrieved in a single query. Finally,
        # another query retrieves the URL for the "more jobs page" link.
```

Reply:
I believe the extra query here is due to [this change](https://docs.wagtail.io/en/stable/releases/2.9.html#sitemiddleware-and-request-site-deprecated) in Wagtail 2.9, which deprecates the SiteMiddleware. See [this documentation change](https://github.com/wagtail/wagtail/pull/5787/files#diff-94d3651c6eca34df52284ef3a2359cbb) for specifics.

To avoid this extra query, we could add `Site.find_for_request(self.request)` to the `setUp` method of the tests. This would pre-cache the extra lookup on the request object.

Reply:
@chosak I fixed JobManager block tests to avoid extra query by updating `setUp` method that you recommended.